packages/api/server/channels/app.mts

```diff
@@ -18,9 +18,16 @@ import { loadApp } from '../../apps/app.mjs';
 import { fileUpdated, pathToApp } from '../../apps/disk.mjs';
 import { vite } from '../../exec.mjs';
 
+const VITE_PORT_REGEX = /Local:.*http:\/\/localhost:([0-9]{1,4})/;
+
 type AppContextType = MessageContextType<'appId'>;
 
-const processes = new Map<string, ChildProcess>();
+type ProcessMetadata = {
+  process: ChildProcess;
+  port: number | null;
+};
+
+const processMetadata = new Map<string, ProcessMetadata>();
```

Comment:
Note that now, `processes` is called `processMetadata` and includes a process plus associated metadata (currently just the port it is running on).